server/src/engineModel/EngineModel.ts

```diff
@@ -92,6 +93,15 @@ export class EngineModel {
                 this.completionProvider.addCompletionItem(method);
             });
         }
+
+        if (!!yclass.instances && !!yclass.instances[0].item) {
```

Comment:
1) This condition could be made clearer via a function.
2) From my point of view, you seem to be assuming that empty arrays are falsey, but according to what I can see on the Web they are truthy, so you can theoretically reach the `yclass.instances[0]` with `instances` holding no element. And if you end up having to add checks for this, that leads us back to (1). ^^
3) Wow, every commit on this extension seems to add logic into which we use `[0]` and ignore the rest. That’s always confusing me. Is it easy to explain with a comment, here?